src/main/scala/fasta.scala

```diff
@@ -127,115 +127,45 @@ case object fasta {
     new DenotationParser(sequence, sequence.label)({ v: String => Some(FastaSequence(v)) })
 
-  /*
-    This method returns an iterator over `Map[String, String]` which can be directly used by `FASTA.parse`.
-
-    **NOTE** the `lines` iterator should *not* be used after calling `parseFromLines` on it.
-  */
-  final def parseMap(lines: Iterator[String]): Iterator[Map[String, String]] = new Iterator[Map[String, String]] {
-
-    // NOTE see https://groups.google.com/forum/#!topic/scala-user/BPjFbrglfMs for why this is that ugly
-
-    // Iterator state
-    var currentHeader : String = ""
-    var nextHeader    : String = ""
-    // this stores the current "next" map
-    var currentMap: Option[Map[String, String]] = None
-    // if you do hasNext twice, it won't call evalNext again
-    var hasBeenChecked: Boolean = false
-    var isFirst       : Boolean = true
-
-    private def evalNext: Option[Map[String, String]] = {
-
-      val currentSequence = new StringBuilder
-
-      var foundHeader = false
-      // first time: find a header, and accumulate seq til you find another header
-      if (isFirst) {
-
-        while (lines.hasNext && nextHeader.isEmpty) {
-
-          val currentLine = lines.next
-
-          if (header is currentLine) {
-
-            if(!foundHeader) {
-              currentHeader = currentLine
-              foundHeader   = true
-            }
-            else {
-              nextHeader = currentLine
-            }
-          }
-          else if (currentHeader.nonEmpty) {
-            currentSequence append currentLine
-          }
-        }
-
-        isFirst = false
-      }
-      else if(nextHeader.nonEmpty) {
-
-        currentHeader = nextHeader
-        nextHeader    = ""
-        foundHeader   = false
-
-        while (lines.hasNext && !foundHeader) {
-
-          val currentLine = lines.next
-
-          if (header is currentLine) {
-
-            foundHeader = true
-            nextHeader  = currentLine
-          }
-          else {
-            currentSequence append currentLine
-          }
-        }
-      }
-
-      if (currentHeader.isEmpty) {
-        None
-      }
-      else {
-        val tmpMap = Some(
-          collection.immutable.HashMap(
-            header.label   -> currentHeader,
-            sequence.label -> currentSequence.toString
-          )
-        )
-        // THIS IS IMPORTANT
-        currentHeader = ""
-        tmpMap
-      }
+  /* Note, all we only use `next`, `hasNext` and `head` method here, so the `lines` iterator can be used after calling any of these extension methods and can be considered as the parsing "remainder" (i.e. unparsed part) */
+  implicit class BufferedIteratorFASTAOps(val lines: BufferedIterator[String]) extends AnyVal {
+
+    /* Similar to `takeWhile`, but the iterator **can be reused** after */
+    private def cutUntil(condition: String => Boolean): Seq[String] = {
+      if (lines.hasNext && !condition(lines.head))
```

Comment:
what optionHead?? this is [`BufferedIterator#head`](http://scala-lang.org/api/2.11.8/index.html#scala.collection.BufferedIterator@head:A)